[15_0_X] Backport: Restructure and comment the tau embedding method

winterchristian opened the pull request:
#### PR description:

This is a backport of https://github.com/cms-sw/cmssw/pull/48408.
This backport is needed to produce 2024 tau embedding samples.

The following is the description of https://github.com/cms-sw/cmssw/pull/48408:

This PR restructures and cleans the [tau embedding method](https://twiki.cern.ch/twiki/bin/viewauth/CMS/TauEmbeddingSamplesUL) ([`TauAnalysis/MCEmbeddingTools`](https://github.com/cms-sw/cmssw/tree/master/TauAnalysis/MCEmbeddingTools)) so that it is possible to produce RUN 3 tau embedding samples with CMS submission workflows.
One goal was to remove the need of `--customise` or `--customise_commands` in the tau embedding `cmsDriver.py` commands.
This was possible due to modifiers and the possibility to add python config fragments to some `cmsDriver.py` steps.
Only the `HLT` step required additional modification of the `cmsDriver.py` [config builder](https://github.com/cms-sw/cmssw/pull/48408/files#diff-cdb26c43130acb8faa9c94529dd55a55671d2621695d7c5a2a47aa20a62e62df).

The other goal of this PR is to have more understandable, better commented code and therefore sustainable code.

#### Tau embedding `cmsDriver.py` commands
With the changes introduced in this PR, tau embedding samples can be produced using the following `cmsDriver.py` commands.

<details>
<summary>Selection</summary>

```shell
cmsDriver.py \
--step RAW2DIGI,L1Reco,RECO,PAT,FILTER:TauAnalysis/MCEmbeddingTools/Selection_FILTER_cff.makePatMuonsZmumuSelection \
--processName SELECT \
--data \
--scenario pp \
--conditions auto:run3_data \
--era Run3_2024 \
--eventcontent TauEmbeddingSelection \
--datatier RAWRECO \
--filein "root://cmsdcache-kit-disk.gridka.de:1094//store/data/Run2024C/Muon0/RAW/v1/000/380/115/00000/00979445-916c-42e2-8038-428d7bd4f176.root" \
--fileout ...
```

</details>
<details>
<summary>LHE and Cleaning</summary>

```shell
cmsDriver.py \
--step USER:TauAnalysis/MCEmbeddingTools/LHE_USER_cff.embeddingLHEProducerTask,RAW2DIGI,RECO:TauAnalysis/MCEmbeddingTools/Cleaning_RECO_cff.reconstruction \
--processName LHEembeddingCLEAN \
--data \
--scenario pp \
--conditions auto:run3_data \
--era Run3_2024 \
--eventcontent TauEmbeddingCleaning \
--datatier RAWRECO \
--procModifiers tau_embedding_mu_to_mu \
--filein ... \
--fileout ...
```

</details>
<details>
<summary>Simulation Gen</summary>

```shell
cmsDriver.py TauAnalysis/MCEmbeddingTools/python/Simulation_GEN_cfi.py \
--step GEN,SIM,DIGI,L1,DIGI2RAW \
--processName SIMembeddingpreHLT \
--mc \
--beamspot DBrealistic \
--geometry DB:Extended \
--era Run3_2024 \
--conditions auto:phase1_2024_realistic \
--eventcontent TauEmbeddingSimGen \
--datatier RAWSIM \
--procModifiers tau_embedding_mu_to_mu \
--filein ... \
--fileout ...
```

</details>
<details>
<summary>Simulation HLT</summary>

```shell
cmsDriver.py \
--step HLT:TauAnalysis/MCEmbeddingTools/Simulation_HLT_customiser_cff.embeddingHLTCustomiser.Fake2 \
--processName SIMembeddingHLT \
--mc \
--beamspot DBrealistic \
--geometry DB:Extended \
--era Run3_2024 \
--conditions auto:phase1_2024_realistic \
--eventcontent TauEmbeddingSimHLT \
--datatier RAWSIM \
--filein ... \
--fileout ...
```

</details>
<details>
<summary>Simulation Reco</summary>

```shell
cmsDriver.py \
--step RAW2DIGI,L1Reco,RECO:TauAnalysis/MCEmbeddingTools/Simulation_RECO_cff.reconstruction,RECOSIM \
--processName SIMembedding \
--mc \
--beamspot DBrealistic \
--geometry DB:Extended \
--era Run3_2024 \
--conditions auto:phase1_2024_realistic \
--eventcontent TauEmbeddingSimReco \
--datatier RAW-RECO-SIM \
--filein ... \
--fileout ...
```

</details>
<details>
<summary>Merging</summary>

```shell
cmsDriver.py \
--step USER:TauAnalysis/MCEmbeddingTools/Merging_USER_cff.merge_step,PAT \
--processName MERGE \
--data \
--scenario pp \
--conditions auto:run3_data \
--era Run3_2024 \
--eventcontent TauEmbeddingMerge \
--datatier USER \
--inputCommands 'keep *_*_*_*' \
--filein ... \
--fileout ...
```

</details>
<details>
<summary>NanoAOD</summary>

```shell
cmsDriver.py \
--step NANO:TauAnalysis/MCEmbeddingTools/Nano_cff.embedding_nanoAOD_seq \
--data \
--scenario pp \
--conditions auto:run3_data \
--era Run3_2024 \
--eventcontent TauEmbeddingNANOAOD \
--datatier NANOAODSIM \
--filein ... \
--fileout ...
```

</details>


#### PR validation:

I have included simple release validation and unit tests for tau embedding 2024.

Commit:
disable InitialRecoCorrection

Commit:
add tau embedding process modifiers

Commit:
first version without customize, but not HLT step

Commit:
rewrite tau embedding HLT step, by adding customise functionality in cmsDriver HLT step

Commit:
add config fragment for embedding NanoAOD step with special NanoAOD embedding table

Commit:
remove not needed python config fragments and switch to a consistent naming

Commit:
remove not needed functionalities, which were used for some tests in the past

Commit:
add comments to filter python config fragment

Commit:
add comments to tau embedding python config fragments

Commit:
move tau embedding header files to interface folder

Commit:
add some documentation to the tau embedding method

Commit:
format code

Commit:
change tests to new structure

Commit:
fix typos

Commit:
refactor MuMuForEmbeddingSelector to simplify the handling of values

Commit:
debug, as the wrong import order breaks the embedding cleaning config

winterchristian committed:
Add custom EventContent for tau embedding samples (#5)

* add special tau embedding EventContents

* add NANOAOD to NanoAOD event content name to be handled properly

* adapt eventcontent in cmsDriver commands

---------

Co-authored-by: Christian Winter <c.winter@kit.edu>

Commit:
fix spelling

Commit:
adapt cmsDriver commands for release validation and add tests for 2022

Commit:
switch to IBEos relval datasets in unittests and reactivate them

cmsbuild commented:
A new Pull Request was created by  @winterchristian for CMSSW_15_0_X.

It involves the following packages:

- Configuration/Applications (**operations**)
- Configuration/EventContent (**operations**)
- Configuration/ProcessModifiers (**operations**)
- Configuration/PyReleaseValidation (**upgrade, pdmv**)
- TauAnalysis/MCEmbeddingTools (**simulation**)


@AdrianoDee, @Moanwar, @antoniovilela, @civanch, @cmsbuild, @davidlange6, @dickychant, @fabiocos, @kpedro88, @mandrenguyen, @mdhildreth, @miquork, @rappoccio, @srimanob, @subirsarkar can you please review it and eventually sign? Thanks.
@Martin-Grunewald, @fabiocos, @makortel, @missirol, @mmusich, @slomeo this is something you requested to watch as well.
@antoniovilela, @mandrenguyen, @rappoccio, @sextonkennedy you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

cmsbuild commented:
cms-bot internal usage<!-- bot cache: {"commits":{"00bbeb14e34bad9ea755e63ce9d293cadb0e6b46":{"files":["TauAnalysis/MCEmbeddingTools/plugins/MuMuForEmbeddingSelector.cc","TauAnalysis/MCEmbeddingTools/python/Nano_cff.py"],"squashed":false,"time":1751531488},"19c149c45fa4d1fb7a7483ac9a8f6e53c292bfc4":{"files":["Configuration/Applications/python/ConfigBuilder.py","TauAnalysis/MCEmbeddingTools/python/Simulation_HLT_customiser_cff.py"],"squashed":false,"time":1751531483},"1e14da0bbd98d2c5e0ed8e9274dba26c8ce1f1e2":{"files":["TauAnalysis/MCEmbeddingTools/python/Cleaning_RECO_cff.py","TauAnalysis/MCEmbeddingTools/python/Simulation_HLT_customiser_cff.py"],"squashed":false,"time":1751531591},"40b73e8cbc6f47fdfdc1361b730fb744bbe224ef":{"files":["TauAnalysis/MCEmbeddingTools/interface/CaloCleaner.h","TauAnalysis/MCEmbeddingTools/interface/CollectionMerger.h","TauAnalysis/MCEmbeddingTools/interface/DoubleCollectionMerger.h","TauAnalysis/MCEmbeddingTools/interface/EmbeddingBeamSpotOnlineProducer.h","TauAnalysis/MCEmbeddingTools/interface/MuonDetCleaner.h","TauAnalysis/MCEmbeddingTools/interface/TrackMergeremb.h","TauAnalysis/MCEmbeddingTools/interface/TrackerCleaner.h","TauAnalysis/MCEmbeddingTools/plugins/CaloCleaner.cc","TauAnalysis/MCEmbeddingTools/plugins/CollectionMerger.cc","TauAnalysis/MCEmbeddingTools/plugins/DoubleCollectionMerger.cc","TauAnalysis/MCEmbeddingTools/plugins/EmbeddingBeamSpotOnlineProducer.cc","TauAnalysis/MCEmbeddingTools/plugins/MuonDetCleaner.cc","TauAnalysis/MCEmbeddingTools/plugins/TrackMergeremb.cc","TauAnalysis/MCEmbeddingTools/plugins/TrackerCleaner.cc"],"squashed":false,"time":1751531522},"4525893e136f417fe5742c70ac21a8edc783453b":{"files":["TauAnalysis/MCEmbeddingTools/python/Selection_FILTER_cff.py"],"squashed":false,"time":1751531509},"4ceafe95050bd0d3b613ae9fdab91aed22543677":{"files":["TauAnalysis/MCEmbeddingTools/python/EmbeddingBeamSpotOnline_cfi.py","TauAnalysis/MCEmbeddingTools/python/EmbeddingHltPixelVerticesProducer_cfi.py","TauAnalysis/MCEmbeddingTools/python/EmbeddingVertexCorrector_cfi.py","TauAnalysis/MCEmbeddingTools/python/LHE_USER_cff.py","TauAnalysis/MCEmbeddingTools/python/Merging_USER_cff.py","TauAnalysis/MCEmbeddingTools/python/Selection_FILTER_cff.py","TauAnalysis/MCEmbeddingTools/python/Simulation_GEN_cfi.py","TauAnalysis/MCEmbeddingTools/python/customisers.py"],"squashed":false,"time":1751531493},"68e8c818ae45be64f6f0c5725aa94f8f056c3b09":{"files":["TauAnalysis/MCEmbeddingTools/docs/README.md"],"squashed":false,"time":1751531551},"8d9352a1a6baaf299b2bbc5bf1c1fb37ba0ecfe4":{"files":["Configuration/EventContent/python/EventContent_cff.py","TauAnalysis/MCEmbeddingTools/docs/README.md","TauAnalysis/MCEmbeddingTools/docs/cmsDriver_cmds_DATA_RUN3.sh","TauAnalysis/MCEmbeddingTools/python/Cleaning_RECO_cff.py","TauAnalysis/MCEmbeddingTools/python/Embedding_EventContent_cff.py","TauAnalysis/MCEmbeddingTools/python/LHE_USER_cff.py","TauAnalysis/MCEmbeddingTools/python/Merging_USER_cff.py","TauAnalysis/MCEmbeddingTools/python/Nano_cff.py","TauAnalysis/MCEmbeddingTools/python/Selection_FILTER_cff.py","TauAnalysis/MCEmbeddingTools/python/Simulation_GEN_cfi.py","TauAnalysis/MCEmbeddingTools/python/Simulation_HLT_customiser_cff.py","TauAnalysis/MCEmbeddingTools/python/Simulation_RECO_cff.py","TauAnalysis/MCEmbeddingTools/test/run_2016postVFPUL_workflow_tests.sh","TauAnalysis/MCEmbeddingTools/test/run_2016preVFPUL_workflow_tests.sh","TauAnalysis/MCEmbeddingTools/test/run_2017UL_workflow_tests.sh","TauAnalysis/MCEmbeddingTools/test/run_2018UL_workflow_tests.sh","TauAnalysis/MCEmbeddingTools/test/run_2022_workflow_tests.sh"],"squashed":false,"time":1751531615},"9764db272b8a388412ac407022cc3581168cbe94":{"files":["TauAnalysis/MCEmbeddingTools/plugins/EmbeddingLHEProducer.cc","TauAnalysis/MCEmbeddingTools/plugins/MuMuForEmbeddingSelector.cc"],"squashed":false,"time":1751531505},"9800f8feffdea6866fb6c28d6e4114e1b6f8c106":{"files":["TauAnalysis/MCEmbeddingTools/test/BuildFile.xml","TauAnalysis/MCEmbeddingTools/test/run_2016postVFPUL_workflow_tests.sh","TauAnalysis/MCEmbeddingTools/test/run_2016preVFPUL_workflow_tests.sh","TauAnalysis/MCEmbeddingTools/test/run_2017UL_workflow_tests.sh","TauAnalysis/MCEmbeddingTools/test/run_2018UL_workflow_tests.sh","TauAnalysis/MCEmbeddingTools/test/run_2022_workflow_tests.sh"],"squashed":false,"time":1751531659},"b1f4c38dea5eb569e91dfcb6a9d207c9c17a4945":{"files":["TauAnalysis/MCEmbeddingTools/python/Cleaning_RECO_cff.py","TauAnalysis/MCEmbeddingTools/python/LHE_USER_cff.py","TauAnalysis/MCEmbeddingTools/python/Merging_USER_cff.py","TauAnalysis/MCEmbeddingTools/python/Nano_cff.py","TauAnalysis/MCEmbeddingTools/python/Selection_FILTER_cff.py","TauAnalysis/MCEmbeddingTools/python/Simulation_GEN_cfi.py","TauAnalysis/MCEmbeddingTools/python/Simulation_HLT_customiser_cff.py","TauAnalysis/MCEmbeddingTools/python/Simulation_RECO_cff.py"],"squashed":false,"time":1751531517},"bb4150c92bcab3f6efd8512a00e118c69cca76d1":{"files":["TauAnalysis/MCEmbeddingTools/python/EmbeddingLHEProducer_cfi.py"],"squashed":false,"time":1751531451},"c19208184ed83789b433b11ae4b1cff65663c6ca":{"files":["TauAnalysis/MCEmbeddingTools/plugins/MuMuForEmbeddingSelector.cc"],"squashed":false,"time":1751531564},"d67a9b5f1bf94ab7478c477d945ff1507693464e":{"files":["TauAnalysis/MCEmbeddingTools/python/Cleaning_RECO_cff.py","TauAnalysis/MCEmbeddingTools/python/EmbeddingLHEProducer_cfi.py","TauAnalysis/MCEmbeddingTools/python/EmbeddingPythia8Hadronizer_cfi.py","TauAnalysis/MCEmbeddingTools/python/EmbeddingVertexCorrector_cfi.py","TauAnalysis/MCEmbeddingTools/python/Merging_cff.py","TauAnalysis/MCEmbeddingTools/python/SelectingProcedure_cff.py","TauAnalysis/MCEmbeddingTools/python/Simulation_RECO_cff.py"],"squashed":false,"time":1751531476},"d958c2a53290fc432738963b90170d1422d8f081":{"files":["TauAnalysis/MCEmbeddingTools/docs/README","TauAnalysis/MCEmbeddingTools/docs/README.md","TauAnalysis/MCEmbeddingTools/docs/cmsDriver_cmds_DATA_RUN3.sh"],"squashed":false,"time":1751531528},"da1d64f822f208eb5dc9088aa58f11019899451c":{"files":["Configuration/Applications/python/ConfigBuilder.py"],"squashed":false,"time":1751531639},"f033371cd8b40935ea5f904fd5f2c40069b755d3":{"files":["Configuration/ProcessModifiers/python/tau_embedding_emu_cff.py","Configuration/ProcessModifiers/python/tau_embedding_etauh_cff.py","Configuration/ProcessModifiers/python/tau_embedding_mu_to_e_cff.py","Configuration/ProcessModifiers/python/tau_embedding_mu_to_mu_cff.py","Configuration/ProcessModifiers/python/tau_embedding_mutauh_cff.py","Configuration/ProcessModifiers/python/tau_embedding_tauhtauh_cff.py","TauAnalysis/MCEmbeddingTools/python/EmbeddingLHEProducer_cfi.py","TauAnalysis/MCEmbeddingTools/python/EmbeddingPythia8Hadronizer_cfi.py","TauAnalysis/MCEmbeddingTools/test/run_2022_workflow_tests.sh"],"squashed":false,"time":1751531459},"f2082679ca8f513dd02f44676b3f0ec8b8c4508a":{"files":["TauAnalysis/MCEmbeddingTools/plugins/MuMuForEmbeddingSelector.cc"],"squashed":false,"time":1751531534},"f2925fbb2242ac3005f1c0c3e6befd76f6dac5d2":{"files":["Configuration/PyReleaseValidation/python/relval_standard.py","Configuration/PyReleaseValidation/python/relval_steps.py"],"squashed":false,"time":1751531646},"ffb61a6caa4343b6b3763e9eb515a0ef64b5a914":{"files":["TauAnalysis/MCEmbeddingTools/plugins/MuMuForEmbeddingSelector.cc","TauAnalysis/MCEmbeddingTools/test/BuildFile.xml","TauAnalysis/MCEmbeddingTools/test/run_2016postVFPUL_workflow_tests.sh","TauAnalysis/MCEmbeddingTools/test/run_2016preVFPUL_workflow_tests.sh","TauAnalysis/MCEmbeddingTools/test/run_2017UL_workflow_tests.sh","TauAnalysis/MCEmbeddingTools/test/run_2018UL_workflow_tests.sh","TauAnalysis/MCEmbeddingTools/test/run_2022_workflow_tests.sh"],"squashed":false,"time":1751531542}},"emoji":{},"last_seen_sha":"9800f8feffdea6866fb6c28d6e4114e1b6f8c106","signatures":{}} -->

civanch commented:
please test

cmsbuild commented:
+1

**Size**: This PR adds an extra 16KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c06c73/47288/summary.html
**COMMIT**: 9800f8feffdea6866fb6c28d6e4114e1b6f8c106
**CMSSW**: CMSSW_15_0_X_2025-07-19-1100/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/48557/47288/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - You potentially [removed 4 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_0_X_2025-07-19-1100+c06c73/69444/validateJR/logRootQA-events.log) from the logs
 - [Reco comparison results: 10 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_0_X_2025-07-19-1100+c06c73/69444/validateJR.html)
 - DQMHistoTests: Total files compared: 50
 - DQMHistoTests: Total histograms compared: 4020943
 - DQMHistoTests: [Total failures: 19](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_0_X_2025-07-19-1100+c06c73/69444/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 4020904
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 49 files compared)
 - Checked 218 log files, 189 edm output root files, 50 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_0_X_2025-07-19-1100+c06c73/69444/triggerResults): no differences found

Commit:
change to more reasonable HLT customisation with '+' separator

Commit:
move the replacement of the PV from the HLT to the RECO step

cmsbuild commented:
Pull request #48557 was updated. @AdrianoDee, @Moanwar, @antoniovilela, @civanch, @cmsbuild, @davidlange6, @dickychant, @fabiocos, @kpedro88, @mandrenguyen, @mdhildreth, @miquork, @rappoccio, @srimanob, @subirsarkar can you please check and sign again.

civanch commented:
+1

winterchristian committed:
New Tau Embedding process modifiers  (#6)

* first try distributing process Modifiers of Cleaning step

* disable finale state radiation for mu->mu embedding

* update center of mass energy for run3

* implement correct ModifierChain for tau embedding process modifiers

* revert some changes

* set all process modifier for tau embeddign cleaning step

* Introduce special cleaning and simulation process modifier

* add process modifiers for sim gen step

* Set process modifiers for reco sim step

* add process modifier for tau embedding merging step
Those "toReplaceWith" modifiers need to be applied as late as possible e.g. in StandardSequences.Reconstruction_Data_cff.
This is to avoid collisions with other modules that are cloned and modified versions from the original modules or "toModify" modifier calls that try to modify the original modules. They throw an exeption if they want to do this on the replaced merger modules.

* add autoNANO option for the tau embedding nanoaod files

* adjust test cmsDriver commands to work with the new process modifiers

---------

Co-authored-by: Christian Winter <c.winter@kit.edu>
(cherry picked from commit 164e69dcc4b608accc38237c695897b69739d0f4)

Commit:
update the cmsDriver commands of the release validation

(cherry picked from commit 2993c3e252fafeb287e2003a1ab6bbbbcede08e4)

Commit:
Adaptation for use with SwitchProducerCUDA

cmsbuild commented:
Pull request #48557 was updated.

civanch commented:
please test

cmsbuild commented:
+1

**Size**: This PR adds an extra 16KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c06c73/48367/summary.html
**COMMIT**: c52227a733db9cb4c52cb6a5775819dd7859b960
**CMSSW**: CMSSW_15_0_X_2025-09-29-2300/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/48557/48367/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - [Reco comparison results: 8 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_0_X_2025-09-29-2300+c06c73/70608/validateJR.html)
 - DQMHistoTests: Total files compared: 51
 - DQMHistoTests: Total histograms compared: 4136246
 - DQMHistoTests: [Total failures: 33](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_0_X_2025-09-29-2300+c06c73/70608/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 4136193
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 50 files compared)
 - Checked 222 log files, 194 edm output root files, 51 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_0_X_2025-09-29-2300+c06c73/70608/triggerResults): no differences found

AdrianoDee commented:
test parameters:

- workflows = 136.901,136.902,136.903,136.904,136.905,136.906
- relvals_opt = --what standard

AdrianoDee commented:
HI @winterchristian, two things:
- given you have also the other backport to 14_2_X could you mention this is a 15_0_X backport in the title?
- I think you will need also the 15_1_X backport to do the full chain from 16_0_X to 14_2_X.

civanch commented:
+1

ftorrresd commented:
enable nano

cmsbuild commented:
+1

**Size**: This PR adds an extra 20KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c06c73/48519/summary.html
**COMMIT**: c52227a733db9cb4c52cb6a5775819dd7859b960
**CMSSW**: CMSSW_15_0_X_2025-10-07-1100/el8_amd64_gcc12
**Additional Tests**: NANO
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/48557/48519/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - You potentially [added 2 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_0_X_2025-10-07-1100+c06c73/70851/validateJR/logRootQA-events.log) to the logs
 - [Reco comparison results: 1 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_0_X_2025-10-07-1100+c06c73/70851/validateJR.html)
 - DQMHistoTests: Total files compared: 51
 - DQMHistoTests: Total histograms compared: 4146542
 - DQMHistoTests: [Total failures: 26](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_0_X_2025-10-07-1100+c06c73/70851/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 4146496
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 50 files compared)
 - Checked 222 log files, 194 edm output root files, 51 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_0_X_2025-10-07-1100+c06c73/70851/triggerResults): no differences found

**Nano size comparison Summary**:

| Sample | kb/ev | ref kb/ev | diff kb/ev | ev/s/thd | ref ev/s/thd | diff rate | mem/thd | ref mem/thd |
| --- | --- | --- | --- | --- | --- | --- | --- | --- |
| 2500.001_NANOmc106Xul16v2 | 3.085 | 3.085 | 0.000 ( +0.0% ) | 3.12 | 2.98 | +4.9% | 2.832 | 2.600 |
| 2500.002_NANOmc106Xul17v2 | 3.144 | 3.144 | 0.000 ( +0.0% ) | 2.98 | 2.83 | +5.4% | 3.170 | 3.014 |
| 2500.003_NANOmc106Xul18v2 | 3.097 | 3.097 | 0.000 ( +0.0% ) | 3.01 | 2.88 | +4.2% | 3.154 | 3.028 |
| 2500.011_NANOdata106Xul16v2 | 1.539 | 1.539 | 0.000 ( +0.0% ) | 5.29 | 4.74 | +11.6% | 2.770 | 2.653 |
| 2500.012_NANOdata106Xul17v2 | 1.925 | 1.925 | 0.000 ( +0.0% ) | 3.53 | 3.31 | +6.6% | 2.932 | 2.833 |
| 2500.013_NANOdata106Xul18v2 | 1.890 | 1.890 | 0.000 ( +0.0% ) | 4.20 | 3.88 | +8.4% | 2.857 | 2.729 |
| 2500.021_NANOmcUL16APVreMINI | 0.022 | 0.022 | 0.000 ( +0.0% ) | 1.52 | 1.36 | +12.3% | 2.921 | 2.564 |
| 2500.022_NANOmcUL16reMINI | 0.022 | 0.022 | 0.000 ( +0.0% ) | 1.47 | 1.40 | +5.1% | 2.913 | 2.561 |
| 2500.023_NANOmcUL17reMINI | 0.022 | 0.022 | 0.000 ( +0.0% ) | 1.27 | 1.20 | +5.7% | 2.636 | 2.517 |
| 2500.024_NANOmcUL18reMINI | 0.022 | 0.022 | 0.000 ( +0.0% ) | 1.15 | 1.10 | +4.9% | 2.854 | 2.703 |
| 2500.031_NANOdataUL16APVreMINI | 0.035 | 0.035 | 0.000 ( +0.0% ) | 1.32 | 1.25 | +6.0% | 2.873 | 2.660 |
| 2500.032_NANOdataUL16reMINI | 0.036 | 0.036 | 0.000 ( +0.0% ) | 1.32 | 1.28 | +3.3% | 2.850 | 2.576 |
| 2500.033_NANOdataUL17reMINI | 0.037 | 0.037 | -0.000 ( -0.1% ) | 1.28 | 1.16 | +10.5% | 2.896 | 2.691 |
| 2500.034_NANOdataUL18reMINI | 0.036 | 0.036 | 0.000 ( +0.0% ) | 1.24 | 1.19 | +4.5% | 2.888 | 2.669 |
| 2500.101_NANOmc130X | 3.220 | 3.220 | 0.000 ( +0.0% ) | 3.55 | 3.34 | +6.1% | 3.107 | 2.954 |
| 2500.111_NANOdata130Xrun3 | 1.799 | 1.799 | 0.000 ( +0.0% ) | 5.35 | 4.67 | +14.5% | 2.809 | 2.686 |
| 2500.131_ScoutingNANOdata130Xrun3 | 0.771 | 0.771 | 0.000 ( +0.0% ) | 27.55 | 27.04 | +1.9% | 1.662 | 1.624 |
| 2500.221_muPOGNANOmc140X | 2.065 | 2.065 | 0.000 ( +0.0% ) | 4.66 | 4.19 | +11.2% | 2.248 | 2.053 |
| 2500.222_EGMNANOmc140X | 3.876 | 3.876 | 0.000 ( +0.0% ) | 3.33 | 3.01 | +10.7% | 2.601 | 2.315 |
| 2500.223_BTVNANOmc140X | 10.607 | 10.607 | 0.000 ( +0.0% ) | 2.11 | 1.90 | +11.1% | 2.819 | 2.501 |
| 2500.224_jmeNANOmc140X | 6.666 | 6.666 | 0.000 ( +0.0% ) | 0.95 | 0.81 | +17.0% | 2.682 | 2.376 |
| 2500.226_lepTrackInfoNANOmc140X | 3.570 | 3.570 | 0.000 ( +0.0% ) | 3.36 | 3.03 | +10.7% | 2.600 | 2.308 |
| 2500.227_ScoutingNANOmc140X | 1.486 | 1.486 | 0.000 ( +0.0% ) | 13.12 | 12.33 | +6.5% | 1.936 | 1.785 |
| 2500.228_ScoutingNANOwithPromptmc140X | 4.474 | 4.474 | 0.000 ( +0.0% ) | 2.81 | 2.67 | +5.3% | 2.911 | 2.540 |
| 2500.231_muPOGNANO140Xrun3 | 1.469 | 1.469 | 0.000 ( +0.0% ) | 6.49 | 6.06 | +7.2% | 2.562 | 2.286 |
| 2500.232_EGMNANOdata140Xrun3 | 2.623 | 2.623 | 0.000 ( +0.0% ) | 4.74 | 4.07 | +16.3% | 2.925 | 2.630 |
| 2500.233_BTVNANOdata140Xrun3 | 5.612 | 5.612 | 0.000 ( +0.0% ) | 3.11 | 2.55 | +21.9% | 3.134 | 2.814 |
| 2500.234_jmeNANOdata140Xrun3 | 3.976 | 3.976 | 0.000 ( +0.0% ) | 1.20 | 1.04 | +15.1% | 2.809 | 2.681 |
| 2500.236_lepTrackInfoNANOdata140Xrun3 | 2.413 | 2.413 | 0.000 ( +0.0% ) | 4.73 | 3.98 | +18.9% | 2.922 | 2.592 |
| 2500.237_ScoutingNANOdata140Xrun3 | 1.101 | 1.101 | 0.000 ( +0.0% ) | 24.21 | 21.53 | +12.4% | 1.656 | 1.483 |
| 2500.238_ScoutingNANOwithPromptdata140Xrun3 | 2.783 | 2.783 | 0.000 ( +0.0% ) | 4.30 | 4.07 | +5.7% | 3.128 | 3.031 |
| 2500.241_l1DPGNANO140Xrun3 | 9.404 | 9.404 | 0.000 ( +0.0% ) | 5.29 | 5.35 | -1.1% | 1.930 | 1.814 |
| 2500.242_muDPGNANO140Xrun3 | 10.377 | 10.377 | 0.000 ( +0.0% ) | 1.20 | 1.17 | +2.3% | 1.739 | 1.669 |
| 2500.243_muDPGNANOBkg140Xrun3 | 6.401 | 6.401 | 0.000 ( +0.0% ) | 9.20 | 9.97 | -7.7% | 1.255 | 1.251 |
| 2500.244_hcalDPGNANO140Xrun3 | 486.016 | 486.016 | 0.000 ( +0.0% ) | 0.94 | 0.80 | +17.9% | 1.709 | 1.609 |
| 2500.245_hcalDPGCalibNANO140Xrun3 | 826.413 | 826.413 | 0.000 ( +0.0% ) | 1.29 | 1.04 | +24.2% | 1.686 | 1.581 |
| 2500.251_hcalDPGMCNANO140Xrun3 | 645.333 | 645.333 | 0.000 ( +0.0% ) | 1.39 | 1.10 | +27.1% | 1.789 | 1.641 |
| 2500.261_NANOmc2024reMINI | 0.021 | 0.021 | 0.000 ( +0.0% ) | 1.26 | 1.07 | +18.0% | 3.083 | 2.704 |
| 2500.271_NANOdata2024reMINI | 0.036 | 0.036 | 0.000 ( +0.0% ) | 1.36 | 1.17 | +16.1% | 2.898 | 2.634 |
| 2500.301_NANOmc150X | 3.217 | 3.217 | 0.000 ( +0.0% ) | 14.68 | 14.53 | +1.0% | 2.714 | 2.641 |
| 2500.311_NANOdata150X | 2.144 | 2.144 | 0.000 ( +0.0% ) | 28.22 | 24.97 | +13.0% | 2.437 | 2.197 |
| 2500.312_NANOdata150X | 2.630 | 2.630 | 0.000 ( +0.0% ) | 22.60 | 18.00 | +25.5% | 2.545 | 2.243 |
| 2500.321_muPOGNANOmc150X | 2.255 | 2.255 | 0.000 ( +0.0% ) | 13.32 | 12.34 | +7.9% | 2.586 | 2.346 |
| 2500.322_EGMNANOmc150X | 4.111 | 4.111 | 0.000 ( +0.0% ) | 13.17 | 10.83 | +21.6% | 2.692 | 2.416 |
| 2500.323_BTVNANOmc150X | 10.552 | 10.552 | 0.000 ( +0.0% ) | 4.21 | 3.98 | +5.8% | 2.912 | 2.619 |
| 2500.324_jmeNANOmc150X | 6.760 | 6.760 | 0.000 ( +0.0% ) | 1.12 | 1.13 | -1.0% | 2.824 | 2.595 |
| 2500.325_jmeNANOrePuppimc150X | 6.763 | 6.763 | 0.000 ( +0.0% ) | 1.04 | 1.10 | -5.1% | 3.143 | 3.121 |
| 2500.326_lepTrackInfoNANOmc150X | 3.721 | 3.721 | 0.000 ( +0.0% ) | 13.29 | 13.99 | -5.0% | 2.693 | 2.445 |
| 2500.327_ScoutingNANOmc150X | 1.562 | 1.562 | 0.000 ( +0.0% ) | 16.09 | 17.28 | -6.9% | 2.457 | 2.298 |
| 2500.328_ScoutingNANOwithPromptmc150X | 4.774 | 4.774 | 0.000 ( +0.0% ) | 9.10 | 9.54 | -4.6% | 2.935 | 2.686 |
| 2500.329_BPHNANOmc150X | 4.109 | 4.109 | 0.000 ( +0.0% ) | 13.78 | 14.83 | -7.1% | 2.747 | 2.539 |
| 2500.331_muPOGNANO150Xrun3 | 1.125 | 1.125 | 0.000 ( +0.0% ) | 25.96 | 22.98 | +13.0% | 2.364 | 2.291 |
| 2500.332_EGMNANOdata150Xrun3 | 2.985 | 2.985 | 0.000 ( +0.0% ) | 21.15 | 23.49 | -10.0% | 2.445 | 2.283 |
| 2500.333_BTVNANOdata150Xrun3 | 7.971 | 7.971 | 0.000 ( +0.0% ) | 4.97 | 5.39 | -7.7% | 2.634 | 2.353 |
| 2500.334_jmeNANOdata150Xrun3 | 4.349 | 4.349 | 0.000 ( +0.0% ) | 1.35 | 1.44 | -6.5% | 2.580 | 2.298 |
| 2500.335_jmeNANOrePuppidata150Xrun3 | 4.332 | 4.332 | 0.000 ( +0.0% ) | 1.19 | 1.25 | -4.9% | 2.694 | 2.560 |
| 2500.336_lepTrackInfoNANOdata150Xrun3 | 2.625 | 2.625 | 0.000 ( +0.0% ) | 22.08 | 25.51 | -13.4% | 2.186 | 2.136 |
| 2500.337_ScoutingNANOdata150Xrun3 | 1.737 | 1.737 | 0.000 ( +0.0% ) | 26.42 | 27.00 | -2.2% | 1.645 | 1.487 |
| 2500.338_ScoutingNANOwithPromptdata150Xrun3 | 2.601 | 2.601 | 0.000 ( +0.0% ) | 14.80 | 15.72 | -5.8% | 2.624 | 2.484 |
| 2500.339_BPHNANOdata150Xrun3 | 2.604 | 2.604 | 0.000 ( +0.0% ) | 28.02 | 30.87 | -9.2% | 2.475 | 2.312 |
| 2500.901_TTbarMINIAOD10.6_UL18v2 | 1.819 | 1.819 | 0.000 ( +0.0% ) | 25.32 | 27.23 | -7.0% | 1.876 | 1.841 |
| 2500.902_TTbarMINIAOD14.0 | 1.665 | 1.665 | 0.000 ( +0.0% ) | 24.64 | 30.10 | -18.1% | 1.461 | 1.429 |
| 2500.911_DYToLL_M-50_13TeV_pythia8 | 14.345 | 14.345 | 0.000 ( +0.0% ) | 5.48 | 7.47 | -26.6% | 1.095 | 1.047 |
| 2500.912_DYToll01234Jets_5f_LO_MLM_Madgraph_LHE_13TeV | 0.310 | 0.171 | 0.140 ( +81.9% ) | 2.05 | 3.38 | -39.3% | 0.851 | 0.851 |
| 2500.913_TTbar_Pow_LHE_13TeV | 0.110 | 0.110 | 0.000 ( +0.0% ) | 2.29 | 2.42 | -5.0% | 0.853 | 0.853 |


## NANO Comparison Summary



**Summary**:
 - You potentially [removed 337 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsNANO/CMSSW_15_0_X_2025-10-07-1100+c06c73/70850/validateJR/logRootQA-events.log) from the logs
 - ROOTFileChecks: Some differences in event products or their sizes found
 - [Reco comparison results: 0 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsNANO/CMSSW_15_0_X_2025-10-07-1100+c06c73/70850/validateJR.html)
 - DQMHistoTests: Total files compared: 22
 - DQMHistoTests: Total histograms compared: 88860
 - DQMHistoTests: [Total failures: 0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsNANO/CMSSW_15_0_X_2025-10-07-1100+c06c73/70850/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 88860
 - DQMHistoTests: Total skipped: 0
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 21 files compared)
 - Checked 134 log files, 80 edm output root files, 22 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsNANO/CMSSW_15_0_X_2025-10-07-1100+c06c73/70850/triggerResults): no differences found
